.github/actions/setup/action.yml

```diff
@@ -0,0 +1,35 @@
+name: "Setup & build"
+description: "Setup node, install dependencies & build"
+runs:
+  using: composite
+  steps:
+    - uses: actions/setup-node@v2
+      with:
+        node-version-file: '.nvmrc'
+    - uses: actions/cache@v2
+      id: build-cache
+      with:
+        path: |
+          packages/bueno/dist
+          packages/headless/dist
+          packages/auth/dist
+          packages/atomic/dist
+          packages/atomic/docs
+          packages/atomic/headless
+          packages/atomic/loader
+          packages/atomic/storybook-static
+          packages/atomic/www
+        key: ${{ github.sha }}
+    - uses: actions/cache@v2
+      with:
+        path: |
+          node_modules
+          */*/node_modules
+        key: ${{ runner.os }}-${{ hashFiles('**/package-lock.json') }}
+    - run: |
+        npm ci
+        npx lerna bootstrap --ci
```

Comment:
> Can't you have a `setup` or `postinstall` script at the top level to do that?

I'm not sure what you mean, every sample I found online used `npm install` the same way., even with setup-node. I also don't see an advantage to moving this elsewhere since it's still only two lines.

> If possible, avoid the use of npx in CI because you then drop the control of the version of the package that you invoke.
> 
> If not possible, fix the version of lerna when invoking it.

I will avoid npx or I will fix the version.

Reply:
For CI/CD, I tend to follow this meme:
![image](https://user-images.githubusercontent.com/12366410/145102880-bda54b4a-bb4d-4a78-bcf1-d07d6576d380.png)

So, what I mean is that I think, if possible, you should use the human command for the CI too.
I reckon that `npm run setup` on the root of the project should work properly because Lerna bootstrap will acquire the CI flag from the environment variable, same for `npm i`.

Reply:
`npm run setup` also runs `npm run lockLernaDependencies`, which is explicitly made to generate `package-lock.json` files. I'd prefer not to generate those.